vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1779,6 +1823,9 @@ func (vm *virtualMachine) setupVirtualMachine(c *govmomi.Client) error {
 	}
 	if vm.template == "" {
 		configSpec.GuestId = "otherLinux64Guest"
+	} else {
+		vdcs := buildVirtualDeviceConfigSpecResize(c, template, vm.hardDisks[0].size)
```

Comment:
can only have a single disk in this case?

Reply:
you can have multiple disks but only one template disk type ex:
```
 disk {
    template = "/nyc/vm/templates/coreos_1506.0.0"
    datastore = "dscluster/datastore2"
    size = "10"
    type = "thin"
  }

  disk {
    name = "test1"
    size = "20"
    type = "thin"
  }
```
Works. If you were to attempt multiple disks such as 
```
  disk {
    template = "/nyc/vm/templates/coreos_1506.0.0"
    datastore = "dscluster/datastore2"
    size = "10"
    type = "thin"
  }

  disk {
    template = "/nyc/vm/templates/coreos_1437.0.0"
    datastore = "dscluster/datastore2"
    size = "10"
    type = "thin"
  }
```
error generated with `[ERROR] Only one bootable disk or template may be given` which is also a limitation on master; https://github.com/terraform-providers/terraform-provider-vsphere/blob/730e48bd327190815b8cd04d0cacbff1d7e70a66/vsphere/resource_vsphere_virtual_machine.go#L855 . This is because template disk is expected to be the boot disk, and can only have one boot disk.